backend/LeetWars.Core/LeetWars.Core.BLL/Services/UserService.cs

```diff
@@ -193,6 +204,101 @@ public async Task<List<UserDto>> GetLeaderBoardAsync(PageSettingsDto? page)
         return _mapper.Map<List<UserDto>>(await users.ToListAsync());
     }
 
+    public async Task<List<UserDto>> GetFriendsLeaderBoardAsync(PageSettingsDto? page)
+    {
+        var currentUser = await GetCurrentUserEntityAsync();

```

Comment:
cannot we reuse the existing method for loading leaderboard users?

Reply:
When using infinite scroll, for example, to retrieve the leaderboard and filter user friends on the frontend, it may work well initially, but it only loads users who are on the current page. So, if all our friends are on the 30th page, they will only appear when we scroll to the end of the leaderboard.

Reply:
but you can extend filter in backend and load only friends